airflow/models.py

```diff
@@ -1945,8 +1945,8 @@ def init_run_context(self, raw=False):
         """
         Sets the log context.
         """
-        self._set_context(self)
         self.raw = raw
+        self._set_context(self)
```

Comment:
raw is used by custom log handler when setting context, e.g. S3TaskHandler use it to determine if it should upload log on close.